src/Components/access-requests-wizard/AccessRequestsWizard.js

```diff
@@ -0,0 +1,284 @@
+          if (details.errors) {
+            throw Error(details.errors.map((e) => e.detail).join('\n'));
+          }
+          if (details && details.target_account) {
+            setInitialValues({
+              [ACCOUNT_NUMBER]: details.target_account,
+              [ORG_ID]: details.target_org,
+              [ACCESS_FROM]: details.start_date,
+              [ACCESS_TO]: details.end_date,
+              [SELECTED_ROLES]: details.roles.map((role) => role.display_name),
+            });
+          } else {
+            throw Error(`Could not fetch details for request ${requestId}`);
+          }
+        }
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        dispatch(
+          addNotification({
+            variant: 'danger',
+            title: 'Could not load access request',
+            description: err.message,
+          })
+        );
+      });
+  }, []);
+
+  const onSubmit = (values) => {
+    setIsSubmitting(true);
+    // https://cloud.redhat.com/docs/api-docs/rbac#operations-CrossAccountRequest-createCrossAccountRequests
+    const body = {
+      target_account: values[ACCOUNT_NUMBER],
+      start_date: values[ACCESS_FROM],
+      end_date: values[ACCESS_TO],
+      target_org: values[ORG_ID],
+      roles: values[SELECTED_ROLES],
+    };
+    setInitialValues(values);
+    apiInstance[isEdit ? 'put' : 'post'](
+      `${API_BASE}/cross-account-requests/${isEdit ? `/${requestId}/` : ''}`,
+      body,
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      }
+    )
+      .then((res) => {
+        if (res.errors && res.errors.length > 0) {
+          throw Error(res.errors[0].detail);
+        }
+        dispatch(
+          addNotification({
+            variant: 'success',
+            title: `${isEdit ? 'Edited' : 'Created'} access request`,
+            description: res.request_id,
+          })
+        );
+        onClose(true);
+      })
+      .catch(({ errors = [] }) => {
+        const isInvalidAccount = /Account .* does not exist/.test(
+          errors[0]?.message || errors[0]?.detail
+        );
+        setError({
+          title: isInvalidAccount
+            ? invalidAccountTitle
+            : `Could not ${variant} access request`,
+          description: isInvalidAccount
+            ? 'Please return to Step 1: Request details and input a new account number for your request.'
+            : errors[0]?.message || errors[0]?.detail,
+        });
+        setIsSubmitting(false);
+      });
+  };
+
+  const validatorMapper = {
+    'validate-account': () => (value) =>
+      value && !error ? undefined : 'Please enter a valid account number',
+    'validate-org-id': () => (value) =>
+      value && !error ? undefined : 'Please enter a valid organization ID',
+    'validate-duration': () => (value) => value?.length > 0,
+  };
+
+  let content = null;
+  if (cancelWarningVisible) {
+    content = (
```

Comment:
Would it be better to return the elements directly in the if/else statements instead of setting them equal to a variable first? I'm just wondering. I'm still very new to React :)

Reply:
@InsaneZein good point, pushed the update. Thank you